plugins/ssr/ssrcommon.ml

```diff
@@ -548,10 +455,18 @@ let pf_abs_evars gl t = pf_abs_evars2 gl [] t
 let ssrautoprop_tac = ref (Proofview.Goal.enter (fun gl -> assert false))
 
 (* Thanks to Arnaud Spiwack for this snippet *)
-let call_on_evar tac e s =
-  let { it = gs ; sigma = s } =
-    tac { it = e ; sigma = s; } in
-  gs, s
+let call_on_evar env sigma tac e =
+  try
+    let tac = Proofview.Unsafe.tclSETGOALS [Proofview.with_empty_state e] <*> tac in
+    let _, init = Proofview.init sigma [] in
+    let name = Names.Id.of_string "legacy_pe" in
+    let (_, final, _, _) = Proofview.apply ~name ~poly:false env tac init in
+    let (gs, final) = Proofview.proofview final in
+    let () = if (gs <> []) then errorstrm (str "Should we tell the user?") in
```

Comment:
Why this? is it not there in the old code, and in any case it is not very informative

Reply:
You mean the error? It was indeed there already, it was just moved around from the previous function.